includes/managers/uploads-manager.php

```diff
@@ -0,0 +1,275 @@
+			$this->file_type_handlers[ $file_type ] = $file_handler;
+
+			// Add the mime type to the allowed mimes list only if unfiltered files upload is enabled.
+			if ( $this->are_unfiltered_files_enabled ) {
+				$this->add_mime_type_to_allowed_mimes_list( $file_type );
+			}
+		}
+	}
+
+	/**
+	 * Extract and Validate Zip
+	 *
+	 * This method accepts a $file array (which minimally should include a 'tmp_name')
+	 *
+	 * @param array $file_path
+	 * @return array|\WP_Error
+	 */
+	public function extract_and_validate_zip( $file_path ) {
+		$validation_result = [];
+
+		/** @var Zip $handler - File Type */
+		$handler = $this->file_type_handlers['zip'];
+
+		// Returns an array of file paths.
+		$extracted = $handler->upload_and_extract_zip( $file_path );
+
+		// If there are no extracted file names, no files passed the extraction validation.
+		if ( empty( $extracted['files'] ) ) {
+			// TODO: Decide what to do if no files passed the extraction validation
+			return new \WP_Error( 'file_error', self::INVALID_FILE_CONTENT );
+		}
+
+		foreach ( $extracted['files'] as $extracted_file ) {
+			// Each file is an array with a 'name' (file path) and 'type' (mime type) properties.
+			$validation_result[] = $this->validate_file( $extracted_file );
+		}
+
+		return $validation_result;
+	}
+
+	/**
+	 * @param array|null $file (If it is a $file array, it must include the 'name' and 'type' properties)
+	 * @return array|\WP_Error
+	 */
+	public function handle_elementor_upload( $file = null ) {
+		// If a file is passed to this method.
+		if ( $file ) {
+			// If $file['fileData'] is set, it signals that the passed file is a Base64 string that needs to be saved
+			// to a file.
+			if ( isset( $file['fileData'] ) ) {
+				$file = $this->save_base64_to_tmp_file( $file );
+
+				if ( is_wp_error( $file ) ) {
+					return $file;
+				}
+			}
+
+			$validation_result = $this->validate_file( $file );
+		} else {
+			$validation_result = [];
+
+			// If no files are passed to this method, check the $_FILES global,
+			foreach ( $_FILES as $file ) {
+				$validation_result[] = $this->validate_file( $file );
+			}
+		}
+
+		return $validation_result;
+	}
+
+	/**
+	 * Save Base64 as File
+	 *
+	 * Saves a Base64 string as a .tmp file in Elementor's temporary files directory.
+	 *
+	 * @param $file
+	 * @return array
+	 */
+	private function save_base64_to_tmp_file( $file ) {
+		$wp_upload_dir = wp_upload_dir();
+		// Get the directory for temporary Elementor uploads.
+		$temp_path = $wp_upload_dir['basedir'] . '/elementor/tmp/';
+		// Create a random temporary file name.
+		$temp_filename = $temp_path . uniqid() . '.tmp';
+
+		$file_content = base64_decode( $file['fileData'] ); // phpcs:ignore
+
+		// If the decode fails
+		if ( ! $file_content ) {
+			return \WP_Error( 'file_error', self::INVALID_FILE_CONTENT );
+		}
+
+		// Save the contents to a temporary file.
+		file_put_contents( $temp_filename, $file_content ); // phpcs:ignore
```

Comment:
Create an external general method for creating temp files and use it wherever you create a temporary file

Reply:
Done. Created `public` method `create_temp_file()`